src/disjoint_set.jl

```diff
@@ -126,9 +127,22 @@ mutable struct DisjointSets{T}
     end
 end
 
+DisjointSets() = DisjointSets{Any}()
+DisjointSets(::Type{T}) where T = DisjointSets{T}()
```

Comment:
This still needs to go.
Passing type in as an arg, rather than a type-param is a thing we used to have to do in Julia 0.3,
but not anymore.

```
julia> Set(Int)
ERROR: MethodError: no method matching length(::Type{Int64})
```

Reply:
Ok, thanks for clarifying.